src/pages/details/DetailsPage.tsx

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import styles from './DetailsPage.module.css';
+import { Product } from '../../models';
+import { fetchDetails, URL } from '../../helpers/api';
+
+interface DetailsProps {
+  id: number | null;
+  onClose: () => void;
+}
+
+function DetailsPage({ id, onClose }: DetailsProps): JSX.Element {
```

Comment:
Deprecated symbol used, consult docs for better alternative: https://discord.com/channels/794806036506607647/1024014517711163503/1260140777011675227